DomainContent/Mexico/onPinataHit.js

```diff
@@ -0,0 +1,182 @@
+        // sound from http://soundbible.com/1952-Punch-Or-Whack.html
+        WHACK_SOUND = SoundCache.getSound("https://hifi-content.s3.amazonaws.com/elisalj/mexico/whack.wav");
+        PARTY_SOUND = SoundCache.getSound("http://mpassets.highfidelity.com/a5f42695-f15a-4f44-9660-14b4f8ca2b29-v1/PartyHorn4.wav");
+    };
+
+    var hits = 0;
+    var hitsToBreak = Math.random() * 10;
+    var TIMEOUT = 30000;
+    var CANDY_LIFETIME = 30;
+
+    var PARTICLE_COUNT = 8;
+
+    var CONFETTI = {
+        type: "ParticleEffect",
+        isEmitting: 1,
+        maxParticles: 8,
+        lifespan: 10,
+        lifetime: 3,
+        emitRate: 100,
+        emitSpeed: 3,
+        speedSpread: 1,
+        script: "(function() { return { preload: function(e) { Script.setTimeout(function(){Entities.editEntity(e, { 'isEmitting': 0, 'script':'' }) },50); } } })",
+        emitOrientation: Quat.fromVec3Degrees({x: 0, y: 90, z: 0}),
+        emitDimensions: {
+            x: 0.5,
+            y: 0.5,
+            z: 0.5
+        },
+        polarStart: Math.PI/180 * 75,
+        polarFinish: Math.PI/180 * (90+15),
+        azimuthStart: -0.3,
+        azimuthFinish: 0.3,
+        emitAcceleration: {
+            x: 0,
+            y: -0.5,
+            z: 0
+        },
+        accelerationSpread: {
+            x: 0,
+            y: 0,
+            z: 0
+        },
+        textures: "http://mpassets.highfidelity.com/a5f42695-f15a-4f44-9660-14b4f8ca2b29-v1/Confetti.png",
+        alpha: 1,
+        alphaSpread: 0,
+        alphaStart: 1,
+        alphaFinish: 1,
+        emitterShouldTrail: 0
+    };
+
+    var CANDY = {
+        name: "Pinata Candy",
+        type: "Model",
+        modelURL: "https://hifi-content.s3.amazonaws.com/elisalj/mexico/candy/candy%20corn.obj",
+        dimensions: {
+            x: 0.0784,
+            y: 0.1164,
+            z: 0.0784
+        },
+        dynamic: true,
+        collisionless: false,
+        shapeType: "Box",
+        gravity: {
+            "x": 0,
+            "y": -9.8,
+            "z": 0
+        },
+        lifetime: CANDY_LIFETIME,
+        script: "https://hifi-content.s3.amazonaws.com/elisalj/emoji_scripts/behaviors/crunch.js"
+    };
+
+    var randColor = function() {
+        return {
+            red: Math.random() * 255,
+            green: Math.random() * 255,
+            blue: Math.random() * 255
+        };
+    };
+
+    var shootConfetti = function(properties) {
+        for (var i = 0; i < PARTICLE_COUNT; i++) {
+            var part = CONFETTI;
+
+            part.colorStart = randColor();
+            part.colorFinish = part.colorStart;
+            part.color = part.colorStart;
+            part.dimensions = {
+                x: Math.random()*0.5,
+                y: Math.random()*0.5,
+                z: Math.random()*0.5
+            };
+            part.position = properties.position;
+            part.rotation = properties.rotation;
+            Entities.addEntity(part, true);
```

Comment:
Does this need a lifetime?

Reply:
Lifetime is included in "CONFETTI" object as 3 seconds.